src/fhirtypes/ElementDefinition.ts

```diff
@@ -524,6 +525,34 @@ export class ElementDefinition {
     }
   }
 
+  /**
+   * Sets flags on this element as specified in a profile or extension.
+   * @todo Add more complete enforcement of rules regarding when these flags can change.
+   * @see {@link http://hl7.org/fhir/R4/profiling.html#mustsupport}
+   * @see {@link http://hl7.org/fhir/R4/elementdefinition-definitions.html#ElementDefinition.mustSupport}
+   * @see {@link http://hl7.org/fhir/R4/elementdefinition-definitions.html#ElementDefinition.isSummary}
+   * @see {@link http://hl7.org/fhir/R4/elementdefinition-definitions.html#ElementDefinition.isModifier}
+   * @param mustSupport - whether to make this element a Must Support element
+   * @param summary - whether to include this element when querying for a summary
+   * @param modifier - whether this element acts as a modifier on the resource
+   * @throws {DisableFlagError} when attempting to disable a flag that cannot be disabled
+   */
+  applyFlags(mustSupport: boolean, summary: boolean, modifier: boolean): void {
+    const disabledFlags = [];
+    if (this.mustSupport && !mustSupport) {
+      disabledFlags.push('Must Support');
+    }
+    if (this.isModifier && !modifier) {
+      disabledFlags.push('Is Modifier');
+    }
+    if (disabledFlags.length) {
+      throw new DisableFlagError(disabledFlags);
+    }
+    this.mustSupport = mustSupport;
+    this.isSummary = summary;
+    this.isModifier = modifier;
```

Comment:
In the same theme, I think we should be setting each of these only if the incoming argument is not `null` or `undefined`.

Reply:
I agree with and will implement your suggestions about only making changes or throwing errors when the arguments are actual `true` or `false` values.